Description: create index file per Debian doc-base requirements
 Debian wants packagers to use doc-base and doc-base requires an index
 file for HTML documents.
Forwarded: not-needed
Author: Hanno Zulla <kontakt@hanno.de>
Last-Update: 2016-02-11

--- a/app/server/ruby/bin/qt-doc.rb
+++ b/app/server/ruby/bin/qt-doc.rb
@@ -41,6 +41,7 @@
 
 docs = []
 filenames = []
+booknames = []
 count = 0
 
 options = {}
@@ -140,7 +141,9 @@
   book_body = book[/<body.*?>/]
   book.gsub!(/<\/?body.*?>/, '')
   book.gsub!(/<meta http-equiv.*?>/, '')
-  File.open("#{qt_gui_path}/book/Sonic Pi - #{name.capitalize}" + (lang != "en" ? " (#{lang})" : "") + ".html", 'w') do |f|
+  bookname = "Sonic Pi - #{name.capitalize}" + (lang != "en" ? " (#{lang})" : "")
+  booknames << bookname
+  File.open("#{qt_gui_path}/book/#{bookname}.html", 'w') do |f|
     f << "<link rel=\"stylesheet\" href=\"../theme/light/doc-styles.css\" type=\"text/css\"/>\n"
     f << "<meta http-equiv=\"Content-Type\" content=\"text/html; charset=utf-8\"/>\n\n"
     f << book_body << "\n"
@@ -291,6 +294,13 @@
   f << "  </qresource>\n</RCC>\n"
 end
 
+File.open("#{qt_gui_path}/book/index.html", 'w') do |f|
+  f << "<link rel=\"stylesheet\" href=\"../theme/light/doc-styles.css\" type=\"text/css\"/>\n"
+  f << "<meta http-equiv=\"Content-Type\" content=\"text/html; charset=utf-8\"/>\n\n"
+  f << "<body><ul>\n"
+  f << booknames.sort.map{|n| "<li><a href=\"#{n}.html\">#{n}</a></li>\n"}.join
+  f << "</ul></body>\n"
+end  
 
 ###
 # Generate info pages
